src/mysql_vm_helpers.py

```diff
@@ -227,6 +233,27 @@ def write_mysqld_config(self, profile: str) -> None:
         with open(f"{MYSQLD_CONFIG_DIRECTORY}/z-custom-mysqld.cnf", "w") as config_file:
             config_file.write(content)
 
+    def setup_logrotate_and_cron(self) -> None:
+        """Create and write the logrotate config file."""
+        logger.debug("Creating logrotate config file")
+
+        with open("templates/logrotate.j2", "r") as file:
+            template = jinja2.Template(file.read())
+
+        rendered = template.render(
+            system_user=MYSQL_SYSTEM_USER,
+            snap_common_directory=CHARMED_MYSQL_COMMON_DIRECTORY,
+            charm_directory=self.charm.charm_dir,
+            unit_name=self.charm.unit.name,
+        )
+
+        with open("/etc/logrotate.d/flush_mysql_logs", "w") as file:
```

Comment:
does the path already exists?
Should we worry if newer ubuntu images stops including logrotate (although sounds very unlikely)

Reply:
It exists for now. I think that the logrotate integration test will fail if it is removed in newer ubuntu versions. I hesitate to add more machinery here than required at the moment